src/main/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersController.java

```diff
@@ -1,7 +1,59 @@
 package uk.gov.hmcts.reform.cpo.controllers;
 
+import io.swagger.annotations.ApiParam;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.cpo.ApplicationParams;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrder;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrderQueryFilter;
+import uk.gov.hmcts.reform.cpo.service.impl.CasePaymentOrdersServiceImpl;
+import uk.gov.hmcts.reform.cpo.validators.annotation.CaseIdsAnnotation;
+
+import org.springframework.validation.annotation.Validated;
+import uk.gov.hmcts.reform.cpo.validators.annotation.IdsAnnotation;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 
 @RestController
+@Validated
+@RequestMapping(value = "/api")
 public class CasePaymentOrdersController {
+
+    private final CasePaymentOrdersServiceImpl casePaymentOrdersServiceImpl;
+    private final ApplicationParams applicationParams;
+
+    public CasePaymentOrdersController(CasePaymentOrdersServiceImpl casePaymentOrdersServiceImpl, ApplicationParams applicationParams) {
+        this.casePaymentOrdersServiceImpl = casePaymentOrdersServiceImpl;
+        this.applicationParams = applicationParams;
+    }
+
+    @GetMapping(value = "case-payment-orders", produces = {"application/json"})
+    public List<CasePaymentOrder> getCasePaymentOrders(@ApiParam(value = "list of ids")
+                                                       @IdsAnnotation @RequestParam("ids") Optional<List<String>> ids,
+                                                       @ApiParam(value = "casesId of ids")
+                                                       @CaseIdsAnnotation @RequestParam("casesId")  Optional<List<String>> casesId,
```

Comment:
Please use the same parameter names as in the https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1478700538#CasePaymentOrders-GetCasePaymentOrders
case-ids
page
size

Reply:
DONE